src/core/image.js

```diff
@@ -506,6 +506,122 @@ var PDFImage = (function PDFImageClosure() {
       }
     },
 
+    /**
+     * Check whether resize image data or not
+     * @param {Number} comps Number of color components, 1 or 3 is supported.
+     * @param {Number} bpc Number of bits per component.
+     * @return {Boolean}
+     */
+    shallResizeImage: function PDFImage_shallResizeImage(comps, bpc) {
+        return (comps === 1 && (bpc === 1 || bpc === 8));
+    },
+
+    /**
+     * Resize black white image data
+     * @param {Uint8Array} imgData image data.
+     * @param {Number} scaleBits
+     * @author Ramsoft
+     */
+    resizeBWPixels: function PDFImage_resizeBWPixels(imgData, scaleBits) {
+      var newWidth = imgData.width >> scaleBits;
+      var newHeight = imgData.height >> scaleBits;
+
+      var newRowBytes = (newWidth + 7) >> 3;
```

Comment:
`+ 7 >> 3` is a bit magical. Please add a comment to explain what this does exactly.